src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -1047,3 +1050,39 @@ void ControllerScriptInterfaceLegacy::softStart(int deck, bool activate, double
     // activate the ramping in scratchProcess()
     m_ramp[deck] = true;
 }
+
+QByteArray ControllerScriptInterfaceLegacy::convertCharset(
+        const ControllerScriptInterfaceLegacy::WellKnownCharsets targetCharset,
+        const QString& value) {
+    switch (targetCharset) {
+    case WellKnownCharsets::LATIN_1:
+    case WellKnownCharsets::ISO_8859_1:
+        return this->convertCharset("ISO-8859-1", value);
+    case WellKnownCharsets::LATIN_9:
+    case WellKnownCharsets::ISO_8859_15:
+        return this->convertCharset("ISO-8859-15", value);
+    case WellKnownCharsets::UCS_2:
+    case WellKnownCharsets::ISO_10646_UCS_2:
+        return this->convertCharset("ISO-10646-UCS-2", value);
+    default:
+        m_pScriptEngineLegacy->throwJSError(QStringLiteral("Unknown charset specified"));
+        return nullptr;
+    }
+}
+
+QByteArray ControllerScriptInterfaceLegacy::convertCharset(
+        const QString& targetCharset, const QString& value) {
+#if QT_VERSION < QT_VERSION_CHECK(6, 4, 0)
+    auto* pCodec = QTextCodec::codecForName(targetCharset.toUtf8());
+    if (!pCodec) {
+        return nullptr;
```

Comment:
just default-construct it (which is even nicely `constexpr`). If you return a nullptr, it will call [`QByteArray(char*, size = -1)`](https://doc.qt.io/qt-6/qbytearray.html#QByteArray-3) instead. Same above and below.
```suggestion
        return QByteArray();
```

Reply:
I'm not sure I'm satisfied with empty array being the marker for an error. Are you sure there isn't a valid use case for returning an empty array? For instance passing an empty string? I know that sounds like a silly use-case bit IDK…

> which is even nicely `constexpr`

WDYM? That I can `return constexpr QByteArray()` or that the compiler automatically `constexpr` a `return QByteArray()`?

Reply:
well, it can knows how to construct the qbytearray inplace while for the other other constructor it needs to jump into the Qt code. See [this example](https://compiler-explorer.com/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:20,endLineNumber:7,positionColumn:20,positionLineNumber:7,selectionStartColumn:20,selectionStartLineNumber:7,startColumn:20,startLineNumber:7),source:'%23include+%3CQByteArray%3E%0A%0AQByteArray+someFunk(bool+cond)+%7B%0A++++if+(cond)+%7B%0A++++++++return+QByteArray()%3B%0A++++%7D%0A++++return+nullptr%3B%0A%7D'),l:'5',n:'1',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:g142,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!((name:qt,ver:'680')),options:'-fPIE+-O2',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+14.2+(Editor+%231)',t:'0')),k:50,l:'4',m:100,n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4) where the `return QByteArray()` is fully inlined while the `return nullptr` is not.